File: 73. Set Matrix Zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        boolean rows[] = new boolean[n];
        boolean cols[] = new boolean[m];
        
        for(int i = 0; i < m; ++i){
            for(int j = 0; j < n; ++j){
                if(matrix[i][j] == 0){
                    rows[j] = true;
                    cols[i] = true;
                }
            }
        }

        for(int i = 0; i  < rows.length; ++i){
            if(rows[i] == true){
                for(int j = 0; j < m; ++j){
                    matrix[j][i] = 0;
                }
            }
        }
        
        for(int i = 0; i  < cols.length; ++i){
            if(cols[i] == true){
                for(int j = 0; j < n; ++j){
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: 73. Set Matrix Zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        
        m = len(matrix)
        n = len(matrix[0])

        rows = [False]*n
        cols = [False]*m

        for i in range(m):
            for j in range(n):
                if(matrix[i][j] == 0):
                    rows[j] = True
                    cols[i] = True
        
        for i in range(len(rows)):
            if(rows[i] == True):
                for j in range(m):
                    matrix[j][i] = 0
            
        
        for i in range(len(cols)):
            if(cols[i] == True):
                for j in range(n):
                    matrix[i][j] = 0